x-pack/plugins/viz_editor/public/common/components/field_list_panel/field_list_panel.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiIcon,
+  // @ts-ignore
+  EuiSuperSelect,
+  ICON_TYPES,
+} from '@elastic/eui';
+// @ts-ignore untyped dependency
+import { palettes } from '@elastic/eui/lib/services';
+import React from 'react';
+import { Datasource } from '../../lib';
+
+interface Props {
+  datasource: Datasource | null;
+}
+
+export function FieldListPanel({ datasource }: Props) {
+  if (!datasource) {
+    return <div />;
+  }
+
+  return (
+    <>
+      {datasource && (
+        <div className="fieldListPanel">
+          {datasource.fields.map(field => (
+            <button
+              type="button"
+              key={field.name}
+              className={`fieldListPanel-field fieldListPanel-field-btn-${field.type}`}
+            >
+              {fieldIcon(field)} <span className="fieldListPanel-field-name">{field.name}</span>
+            </button>
+          ))}
+        </div>
+      )}
+    </>
+  );
+}
```

Comment:
Please try to grab the latest code from the index pattern panel and merge it in here- I can see some argument for fields not being re-sorted for something like a CSV, but they should probably still be searchable in most cases

Reply:
At least in the current POC a header row is required so even reordering shouldn't be a problem